sys/Makefile.dep

```diff
@@ -324,6 +335,13 @@ ifneq (,$(filter sema_inv,$(USEMODULE)))
   USEMODULE += atomic_utils
 endif
 
+ifneq (,$(filter sema,$(USEMODULE)))
+  ifneq (,$(filter ztimer_xtimer_compat,$(USEMODULE)))
+    # sema_timed required 64bit
+    USEMODULE += ztimer64_xtimer_compat
+  endif
+endif
```

Comment:
With #17115 this dependency can be moved to only `tests/sema` and `posix_semaphore`

Reply:
lets wait for it